test/Dialect/Comb/errors.mlir

```diff
@@ -11,3 +11,11 @@ hw.module @err(in %a: i1, in %b: i1) {
   // expected-error @+1 {{Truth tables support a maximum of 63 inputs}}
   %0 = comb.truth_table %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b, %a, %b -> [false]
 }
+
+// -----
+hw.module @err(in %a: i4, in %b: i7) {
+  // expected-note @+1 {{prior use here}}
+  %0 = comb.concat %a, %a : i4, i4
+  // expected-error @+1 {{use of value '%0' expects different type than prior uses: 'i7' vs 'i8'}}
+  %1 = comb.and %0, %b : i7
+}
```

Comment:
Super-Nit: could specify b as an output port and do `hw.output %0` to not involve any other comb ops than concat.

I think what we also want to test is that the return type is not just set by the parser, but the property is also verified when using the generic format or C++ builder function.

Reply:
> Super-Nit: could specify b as an output port and do `hw.output %0` to not involve any other comb ops than concat.

smart :brain:

Reply:
> I think what we also want to test is that the return type is not just set by the parser, but the property is also verified when using the generic format or C++ builder function.

Added a test in generic syntax!